src/grammar/expressions.rs

```diff
@@ -44,22 +67,22 @@ fn expr_bp(p: &mut Parser, min_bp: u8) {
         }
         TokenKind::LParen => {
             p.bump_any();
-            expr_bp(p, 0);
+            let _ = expr_bp(p, 0);
```

Comment:
This may be a small issue, but the return type `Result` is not evaluated here. In case there is an `Err` it's not propagated to the caller, therefore it's ignored. In other places it's also ignored. In case it should be handled one option is to use either a `match` statement or use the short hand operator `?`. The latter propagates the error to the caller (also up the stack) until it is handled lower on the callstack.

Changing the line to

```rust
expr_bp(p, 0)?;
```

should be fine.

Reply:
Added it :+1: